### ISSUE TITLE
QoL suggestions

### ISSUE THREAD

**Issue by csdf27** [Vroonsh/DirectOutput#6]
1.  If would be very helpful to expand the hierachy, such that you have
TableName [RomName]
 |_ Standard effects
    |_ Coin in
    |_ Start button
    |_ Ball at plunger
    |_ Slingshots
    |_ Bumpers
    |_ etc
 |_ Other effects
Most tables share many elements so it would be nice to have them all in the same place.  And I tend to think of effects in terms of what triggers them (e.g. "left slingshot"), not the switch number itself.
Then as we build effects, we insert them into the correct portion of the hierarchy.  Designing it this way also helps you spot if you've missed out a part of the table.  That's much harder to do if all you see is switch, solenoid and lamp numbers.
2. Equally, it would be nice to be able to add a "description" to each Table Element, for example
 Switch W54  (Top left ramp): 7 effects
3. It would be good if Switches, Lamps and Solenoid triggers used a different icon, just to help tell them apart

**Comment by Vroonsh** [comment]
Hey tks for the suggestions !!

1. Do you mean display effects more by outputs category than per switch/element... ?
    If so it could be doable.
    
Otherwise the others could be done i'll have a look

**Comment by csdf27** [comment]
Yes - so that you can choose to categorise effects by what has triggered them.  This is more intuitive than simply using a cryptic switch, solenoid or lamp number.  Usually when we edit table DOF sets we want to edit "what happens when I press the Start Button", not "what happens when switch W13 is activated", also because on one table it might be W13 and on another it might be W54 or whatever.  To do this I guess you'd need to allow each trigger to have a "nicename" which can be displayed instead of (or as well as) the trigger.
Some other ideas that have occurred:
When we start up, can we have a checkbox on the Configuration window that first pops up that says "make these choices the default".  If checked, in future DOTK will skip this window and launch straight to the main window.  Then if the user wants to see that window again, there is a preference in the main window to enable it at next launch.
Similarly, it's annoying to have to go through the "your config file is out of date, do you want to download a new one" every time.  99% of the time I don't, so it would be good to disable this by default and add a preference to enable it.

**Comment by Vroonsh** [comment]
> Yes - so that you can choose to categorise effects by what has triggered them. This is more intuitive than simply using a cryptic switch, solenoid or lamp number. Usually when we edit table DOF sets we want to edit "what happens when I press the Start Button", not "what happens when switch W13 is activated", also because on one table it might be W13 and on another it might be W54 or whatever. To do this I guess you'd need to allow each trigger to have a "nicename" which can be displayed instead of (or as well as) the trigger.

So it's more setting up a element code -> nice name matching table, saved into the dotk file that you want or more some items usual categories (maybe some already setup like bumpers, buttons, slingshot...then the possibility to add some like some specific mech parts) ?

> Similarly, it's annoying to have to go through the "your config file is out of date, do you want to download a new one" every time. 99% of the time I don't, so it would be good to disable this by default and add a preference to enable it.

It could be dangerous if you are currently working on a table which was updated online, you'll miss some commands when exporting your new output lines.

**Comment by csdf27** [comment]
For the latter point - this is not a use case for me.  I use DOTK to test my own custom DOF sets, which when they're ready I will use instead of whatever is online.    For 90% of tables, the existing code is some very basic DOF, so I don't need it or use it.
